mewarpx/mewarpx/utils_store/init_restart_util.py

```diff
@@ -89,6 +90,98 @@ def clean_old_checkpoints(checkpoint_directory="diags",
         shutil.rmtree(dirpath)
 
 
+def _eval_corrupt_checkpoint(checkpoint_dir, checkpoint):
+    """Determine if a checkpoint appears to be corrupt by checking if
+    fluxdata.ckpt is present.
+
+    Arguments:
+        checkpoint_dir (str): Look in this directory for checkpoint
+            directories.
+        checkpoint (str): Checkpoint folder
+
+    Returns:
+        checkpoint_ok (bool): True if the checkpoint appears good
+        (fluxdata.ckpt is present), False if fluxdata.ckpt is missing.
+    """
+    if not os.path.isfile(
+        os.path.join(checkpoint_dir, checkpoint, "fluxdata.ckpt")
+    ):
+        logger.warning(
+            f"Checkpoint {checkpoint} does not contain a flux diag "
+            "checkpoint."
+        )
+        return False
+
+    return True
+
+
+def _remove_corrupt_checkpoint(checkpoint_dir, checkpoint):
+    """Remove a checkpoint that has already been determined to be corrupt and
+    appropriate to remove.
+
+    Arguments:
+        checkpoint_dir (str): Look in this directory for checkpoint
+            directories.
+        checkpoint (str): Checkpoint folder
+    """
+    dirpath = os.path.join(checkpoint_dir, checkpoint)
+    logger.info(f"Removing corrupt checkpoint {dirpath}")
+    shutil.rmtree(dirpath)
+
+
+def _handle_corrupt_checkpoints(checkpoint_dir, checkpoint_list):
+    """Utility function to remove the last checkpoint if it appears to be
+    corrupt. Error out if two or more checkpoints are corrupt.
+
+    Note:
+        A checkpoint is judged corrupt if a flux diag checkpoint does not exist
+        (which is always written after the checkpoint). If runs are ever used
+        without flux diagnostics this logic should be adapted.
+
+    Arguments:
+        checkpoint_dir (str): Look in this directory for checkpoint
+            directories.
+        checkpoint_list (list): List of the checkpoints returned by
+            get_sorted_checkpoints.
+
+    Returns:
+        checkpoint_list (list): The same checkpoint_list, but if the final
+        checkpoint was corrupt it will be removed. If multiple appear to be
+        corrupt an error is raised instead of returning.
+    """
+    last_checkpoint_ok = _eval_corrupt_checkpoint(
+        checkpoint_dir, checkpoint_list[-1]
+    )
+    # Short-circuit the corrupt handling logic: We don't have an issue, carry
+    # on as before.
+    if last_checkpoint_ok:
+        return checkpoint_list
+
+    # If there's only one checkpoint and we know it's corrupt, remove it and
+    # then execute the non-restart logic by returning an empty checkpoint
+    # list.
+    if len(checkpoint_list) == 1:
+        _remove_corrupt_checkpoint(checkpoint_dir, checkpoint_list[-1])
+        return []
```

Comment:
There's an edge case here of jobs never making progress, because they're killed too soon. That's a meta-issue to this PR, though, since there's nothing we can do about it directly at this level.

Rather, we may want to keep a count of whether this case is occurring repeatedly, and either abort the run or automatically shorten the checkpoint interval to compensate

Does AWS expose a variable for which 'attempt' at a job a given execution represents?

Reply:
`AWS_BATCH_JOB_ATTEMPT` environment variable tracks the retry number. https://docs.aws.amazon.com/batch/latest/userguide/job_retries.html